### PR TITLE
api: user-access

### PR DESCRIPTION
**Pull request by jstar1**
Adds user-access along with caching module and test for user-access api

### CODE REVIEW

File: lib/station-log.js [hunk 1]
```diff
@@ -10,26 +10,14 @@ module.exports = {
...
 		var station = {
+			badge: null,
 			user: null,
 			status: 'disabled'
 		};
 		_stationObj[machine] = station;
 		return _stationObj[machine];
 	},
 
-	updateMachine: (machine, setState) => {
```
[the remark this answers is not in the document]
Comment (jstar1) [reply lib/station-log.js]: @bsm5 my fault I forgot to handle this. I'll take care it first thing in the morning.

File: routes/api.js [hunk 2]
```diff
@@ -1,32 +1,270 @@
...
+        'CN' : 'localhost:3001'
+      }
+    }
+  };
+  db.getStation(headerObj['station-id']).then(function(station) {
+    if(!station) {
+      return res.sendStatus(403);
+    } 
+    else if (!station.registered) {
+      if (station.certCN !== cert.subject.CN) {
```
[the remark this answers is not in the document]
Comment (jstar1) [reply routes/api.js]: Alright, I'll put in a patch to handle this @bsm5

[the remark this answers is not in the document]
Comment (mongolsamurai) [reply routes/api.js]: I think all that's needed to fix this is to move the 403 on line 43 to the next-outer block (i.e. swap it with the line below) so it gets run whether or not the cert matches

File: routes/api.js [hunk 3]
```diff
@@ -1,32 +1,270 @@
...
-    } else {
-        res.set({
-            'Content-Type': 'text/plain',
-            'User-Id-String': req.body.id,
-            'Station-State': stationState
+        return res.sendStatus(403);
+      } 
+    }
+    else if (station.certCN !== cert.subject.CN) {
+      //TODO change this to cert.subject.CN
```
Comment (mongolsamurai) [inline routes/api.js]: Looks like this TODO has been resolved?

File: routes/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jstar1) [reply routes/api.js]:
```
 else if (!station.registered) {
      if (station.certCN !== cert.subject.CN) {
        db.modifyStation(headerObj['station-id'], {certCN: cert.subject.CN}).then(function(response) {
          db.logEvent('station management', "Heartbeat for unregistered station with ID '" + headerObj['station-id'] + "' provided a new station CN. Changing to '" + cert.subject.CN + "'");
        });
        return res.sendStatus(403);
      } 
```

Does this not handle the situation where the station is unregistered and station.certCN is not equal to cert.subject.CN

File: lib/station-log.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jstar1) [reply lib/station-log.js]: I'm a bit confused here actually. Do you want me to implement a updateMachine?

File: routes/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jstar1) [reply routes/api.js]: Also I 'm not 100%  sure on how the certs works. If a machine is not registered, can it have a cert?

[the remark this answers is not in the document]
Comment (bsm5) [reply routes/api.js]: This does handle the situation if a station is unregisted and the station.certCN is not equal to the cert.subject.CN.  What this function doesn't handle is returning a returnStatus if the station is unregistered but the staiton.certCN === cert.subject.CN.

Like Andy was saying if you move the: return res.sendStatus(403) to the next outer block you should be fine.  You could also just add an additional else statement that just returns res.sendStatus(403)

File: lib/station-log.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bsm5) [reply lib/station-log.js]: I commented on slack about this.  Let me know if you have any more questions?

**Pull request merged**